pkg/parcacol/querier.go

```diff
@@ -312,22 +352,27 @@ func (q *Querier) QueryRange(
 		)
 	}
 
-	timestampColumnIndex := 0
-	timestampColumnFound := false
-	valueColumnIndex := 0
-	valueColumnFound := false
+	type columnIndex struct {
+		index int
+		found bool
+	}
+	// Add necessary columns and their found value is false by default.
+	columnIndices := map[string]columnIndex{
```

Comment:
@metalmatze how come this is not a struct? I get the convenience of the map access below, but I think we give up compile-time checks that we're actually accessing the correct fields, and not fields that may be unexpected. A switch/case setting the appropriate field of a struct would allow for less mistakes I think.